search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java

```diff
@@ -81,22 +83,11 @@ public SearchQuery toSearchQuery(final UserTaskFilter filter) {
               not(hasChildQuery("taskVariable", taskVarNameQuery)));
 
       // Combine taskVariable, processVariable, and subprocessVariable queries with OR logic
-      variablesQuery = or(taskVariableQuery, processVariableCondition);
+      queries.add(or(taskVariableQuery, processVariableCondition));
     }
+    ofNullable(exists("flowNodeInstanceId")).ifPresent(queries::add); // Default to task
```

Comment:
🔧 I think you can add this search condition directly, right? Without the nullable part of it.
```suggestion
    queries.add(exists("flowNodeInstanceId")); // Default to task
```